### PR TITLE
NOISSUE - messaging in UI

### PR DESCRIPTION
**Pull request by blokovi**
Add messaging in UI

### CODE REVIEW

**Review by drasko — COMMENTED** [review]
This looks good @blokovi, great work!

File: ui/src/Connection.elm [hunk 1]
```diff
@@ -255,4 +255,4 @@ expectResponse toMsg =
                     Err (Http.BadStatus metadata.statusCode)
 
                 Http.GoodStatus_ metadata _ ->
-                    Ok metadata.statusCode
+                    Ok metadata.statusCode
```
Comment (drasko) [inline ui/src/Connection.elm]: Add newline at the end of the file.

File: ui/src/Message.elm [hunk 2]
```diff
@@ -32,29 +41,39 @@ initial =
...
+    , things = Thing.initial
+    , channels = Channel.initial
+    , thingid = ""
     }
 
 
 type Msg
     = SubmitMessage String
-    | SubmitToken String
+      -- | SubmitToken String
```
Comment (drasko) [inline ui/src/Message.elm]: Remove dead code

File: ui/src/Message.elm [hunk 3]
```diff
@@ -32,29 +41,39 @@ initial =
...
     | SentMessage (Result Http.Error Int)
+    | ThingMsg Thing.Msg
+    | ChannelMsg Channel.Msg
+    | SelectedThing String
 
 
-update : Msg -> Model -> ( Model, Cmd Msg )
-update msg model =
+
+-- type ThingId
```
Comment (drasko) [inline ui/src/Message.elm]: Remove dead code.

File: ui/src/Message.elm [hunk 4]
```diff
@@ -32,29 +41,39 @@ initial =
...
         SubmitMessage message ->
             ( { model | message = message }, Cmd.none )
 
         SubmitChannel channel ->
             ( { model | channel = channel }, Cmd.none )
 
-        SubmitToken token ->
-            ( { model | token = token }, Cmd.none )
-
+        -- SubmitToken token ->
```
Comment (drasko) [inline ui/src/Message.elm]: Remove dead code.

File: ui/src/Message.elm [hunk 5]
```diff
@@ -76,6 +95,28 @@ update msg model =
...
+
+        ChannelMsg subMsg ->
+            let
+                ( updatedChannel, channelCmd ) =
+                    Channel.update subMsg model.channels token
+            in
+            ( { model | channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
+
+        SelectedThing thingid ->
+            -- ( Debug.log "model: " { model | thingid = thingid }, Cmd.none )
```
Comment (drasko) [inline ui/src/Message.elm]: Remove dead code.

File: ui/src/Message.elm [hunk 6]
```diff
@@ -87,10 +128,11 @@ view model =
                         [ Form.label [ for "chan" ] [ text "Channel id" ]
                         , Input.email [ Input.id "chan", Input.onInput SubmitChannel ]
                         ]
-                    , Form.group []
-                        [ Form.label [ for "token" ] [ text "Thing token" ]
-                        , Input.text [ Input.id "token", Input.onInput SubmitToken ]
-                        ]
+
+                    -- , Form.group []
```
Comment (drasko) [inline ui/src/Message.elm]: Same here.

File: ui/src/Message.elm [hunk 7]
```diff
@@ -99,10 +141,76 @@ view model =
...
         , Helpers.response model.response
         ]
 
 
+genThingRows : List Thing.Thing -> List (Table.Row Msg)
+genThingRows things =
+    List.map
+        (\thing ->
+            Table.tr []
+                -- [ Table.td [] [ Checkbox.checkbox [ Checkbox.id thing.id ] (Helpers.parseName thing.name) ]
```
Comment (drasko) [inline ui/src/Message.elm]: Same here

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/src/Message.elm [hunk 8]
```diff
@@ -76,6 +95,28 @@ update msg model =
...
+            ( { model | things = updatedThing }, Cmd.map ThingMsg thingCmd )
+
+        ChannelMsg subMsg ->
+            let
+                ( updatedChannel, channelCmd ) =
+                    Channel.update subMsg model.channels token
+            in
+            ( { model | channels = updatedChannel }, Cmd.map ChannelMsg channelCmd )
+
+        SelectedThing thingid ->
```
Comment (darkodraskovic) [inline ui/src/Message.elm:104]: Please move this message to `Thing.elm` module and add `.selectedThingId` to `Thing.Model`. Also, in `Thing.elm` module, use `retrieve` by importing it from `Channel` and move `Channel.RetrieveChannelsForThing` in `Thing.elm`.

File: ui/src/Message.elm [hunk 9]
```diff
@@ -99,10 +141,76 @@ view model =
                     ]
                 ]
             ]
+        , Grid.row []
+            [ Grid.col []
+                [ Html.map ThingMsg
+                    (Grid.row []
+                        [ Grid.col [] [ Input.text [ Input.placeholder "offset", Input.id "offset", Input.onInput Thing.SubmitOffset ] ]
```
Comment (darkodraskovic) [inline ui/src/Message.elm]: Use `Helpers.genFormField` instead of manual generation of `Grid.Col`.

**Review by darkodraskovic — CHANGES_REQUESTED** [review]

File: ui/src/Message.elm (hunk 8 above)
Comment (blokovi) [reply ui/src/Message.elm:104]: I did this but then code is too complicate, so I want to stay with my method of handling this.

**Review by darkodraskovic — APPROVED** [review]
LGTM

**Pull request merged**